node/src/components/fetcher/event.rs

```diff
@@ -67,6 +69,22 @@ pub enum Event<T: Item> {
     TimeoutPeer { id: T::Id, peer: NodeId },
 }
 
+impl<T: Item> Event<T> {
+    pub fn from_get_response_serialized_item(peer: NodeId, serialized_item: &[u8]) -> Option<Self> {
```

Comment:
@fizyk20 Wasn't sure how to handle the server side of the fetcher using a parameter, but I managed to abstract the client side.